File: sekigae.erl
-module(sekigae).
-export([main/0]).

main() ->
    L = [okkun, kitak, keoken, gussan],
    lists:foreach(fun(N) -> io:format("~p~n", [N]) end, shuffle(L)).

shuffle(List) ->
    random:seed(os:timestamp()),
    L = [{random:uniform(), X} || X <- List],
    [X || {_,X} <- lists:sort(L)].

File: sekigae.ml
(* Random.int returns [0, 3) *)
(* ((Random.int 3) - 1) returns -1 or 0 or 1. *)
(* $ ocamlc -o sekigae sekigae.ml *)
(* $ ./sekigae *)

Random.self_init()
let shuffle = Array.sort (fun _ _ -> (Random.int 3) - 1)
let ary = [|"okkun"; "kitak"; "keoken"; "gussan"|]
let _ = shuffle ary
let () = Array.iter (Printf.printf "%s ") ary
